### PR TITLE
Task/mint 3162 mto card be

### PR DESCRIPTION
**Pull request by StevenWadeOddball**
# MINT-3162, [MINT-3191](https://jiraent.cms.gov/browse/MINT-3191), [MINT-3190](https://jiraent.cms.gov/browse/MINT-3190), MINT-3189, [MINT-3188](https://jiraent.cms.gov/browse/MINT-3188)

## Description

1. Add Lockable section
2. Add RecentEdit resolver
3. Add parent level status resolver
4. Add milestone resolver

Note, both 2 and 3 are subject to change based on further implementation (change history and the marked as ready for review status)

## How to test this change

1.  Use Postman to
3. Create a Model Plan
5. Query the status and recent edits using postman
   a. should show ready to start and nil
6. Add a category and query it again. You should see status in progress and date should be the created date of the category
7. Do the same with a milestone
8. Recent edit should update to the milestone date.
9. Date should update upon editing as well.
10. Query Milestones on the MTOMatrix level with postman.

## PR Author Checklist

- [ ] I have provided a detailed description of the changes in this PR.
- [ ] I have provided clear instructions on how to test the changes in this PR.
- [ ] I have updated tests or written new tests as appropriate in this PR.
- [ ] Updated the [Postman Collection](../MINT.postman_collection.json) if necessary.

## PR Reviewer Guidelines

- It's best to pull the branch locally and test it, rather than just looking at the code online!
- When approving a PR, provide a reason _why_ you're approving it
  - e.g. "Approving because I tested it locally and all functionality works as expected"
  - e.g. "Approving because the change is simple and matches the Figma design"
- Don't be afraid to leave comments or ask questions, especially if you don't understand why something was done! (This is often a great time to suggest code comments or documentation updates)
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.

### CODE REVIEW

**Review by ClayBenson94 — CHANGES_REQUESTED** [review]
Some nitpicks and suggestions to simplify stuff, so nothing really critical here.

If you want to just either leave things as-is, implement changes, or just leave some `todo` comments, any combination of those is fine by me!

File: pkg/models/base_struct.go [hunk 1]
```diff
@@ -62,3 +66,36 @@ func (b baseStruct) GetModifiedBy() *string {
...
+			return -1
+		} else if time2.After(time1) {
+			return 1
+		}
+		return 0 // Default comparison if times are equal
+	})
+
+	return baseStructs[0].MostRecentModification()
+
+}
```
[the remark this answers is not in the document]
Comment (StevenWadeOddball) [reply pkg/models/base_struct.go]: Thank Clay! Yes that's a great call! Thanks for sharing the app code for it.

File: pkg/graph/schema/types/mto.graphql [hunk 2]
```diff
@@ -3,9 +3,23 @@
...
 }
 
+type RecentModification {
+  modifiedBy: UUID
+  modifiedByUserAccount: UserAccount
+  modifiedDts: Time
+}
+
+enum MTOStatus {
+  READY_TO_START
```
[the remark this answers is not in the document]
Comment (StevenWadeOddball) [reply pkg/graph/schema/types/mto.graphql]: I appreciate it! I also updated the Review Enum to match the task list sections

File: pkg/graph/resolvers/mto.go [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+	// 	return models.MTOStatusReadyToStart, err
+	// }
+	// if len(solutions) > 0 {
+	// 	return models.MTOStatusInProgress, nil
+	// }
+
+	//If there are no categories, milestones, or solutions, it is just ready to start
+	return models.MTOStatusReadyToStart, nil
+
+}
```
[the remark this answers is not in the document]
Comment (StevenWadeOddball) [reply pkg/graph/resolvers/mto.go]: Agreed! Thanks a bunch! I implemented it!

**Review by ClayBenson94 — APPROVED** [review]
Looks good Steven!

There's currently a failing E2E test, but guessing it's a flaky one given it's for Data Exchange, and it's not failing on the [feature/MINT-3175_mto](https://github.com/CMS-Enterprise/mint-app/tree/feature/MINT-3175_mto) base branch

**Pull request merged**